test/e2e_env/multizone/localityawarelb/locality_multizone_hybrid.go

```diff
@@ -0,0 +1,254 @@
+	return fmt.Sprintf(`
+type: ExternalService
+mesh: "%s"
+name: "%s"
+tags:
+  kuma.io/service: "%s"
+  kuma.io/protocol: http
+  kuma.io/zone: "%s"
+networking:
+  address: "%s"
+`, mesh, name, name, zone, net.JoinHostPort(ip, "8080"))
+}
+
+func InstallExternalService(name string) InstallFunc {
+	return func(cluster Cluster) error {
+		return cluster.DeployApp(
+			WithArgs([]string{"test-server", "echo", "--port", "8080", "--instance", name}),
+			WithName(name),
+			WithoutDataplane(),
+			WithVerbose())
+	}
+}
+
+func ExternalServicesWithLocalityAwareLb() {
+	const mesh = "external-service-locality-lb"
+	const meshNoZoneEgress = "external-service-locality-lb-no-egress"
+	const namespace = "external-service-locality-lb"
+
+	BeforeAll(func() {
+		// Global
+		Expect(NewClusterSetup().
+			Install(YamlUniversal(MeshMTLSOnAndZoneEgressAndNoPassthrough(mesh, "true"))).
+			Install(YamlUniversal(MeshMTLSOnAndZoneEgressAndNoPassthrough(meshNoZoneEgress, "false"))).
+			Setup(env.Global)).To(Succeed())
+
+		E2EDeferCleanup(func() {
+			Expect(env.Global.DeleteMesh(mesh)).To(Succeed())
+		})
+		Expect(WaitForMesh(mesh, env.Zones())).To(Succeed())
+
+		// Universal Zone 1
+		E2EDeferCleanup(func() {
+			Expect(env.UniZone1.DeleteMeshApps(mesh)).To(Succeed())
+		})
+		Expect(NewClusterSetup().
+			Install(DemoClientUniversal(
+				"uni-zone4-demo-client",
+				mesh,
+				WithTransparentProxy(true),
+			)).
+			Install(DemoClientUniversal(
+				"uni-zone4-demo-client-no-egress",
+				meshNoZoneEgress,
+				WithTransparentProxy(true),
+			)).
+			Install(InstallExternalService("external-service-in-uni-zone4")).
+			Install(InstallExternalService("external-service-in-kube-zone1")).
+			Install(InstallExternalService("external-service-in-both-zones")).
+			Setup(env.UniZone1),
+		).To(Succeed())
+
+		// Kubernetes Zone 2
+		E2EDeferCleanup(func() {
+			Expect(env.KubeZone1.TriggerDeleteNamespace(namespace)).To(Succeed())
+		})
+		Expect(NewClusterSetup().
+			Install(NamespaceWithSidecarInjection(namespace)).
+			Install(DemoClientK8s(mesh, namespace)).
+			Setup(env.KubeZone1)).ToNot(HaveOccurred())
+
+		Expect(NewClusterSetup().
+			Install(YamlUniversal(zoneExternalService(mesh, env.UniZone1.GetApp("external-service-in-uni-zone4").GetIP(), "external-service-in-uni-zone4", "kuma-4"))).
+			Install(YamlUniversal(zoneExternalService(mesh, env.UniZone1.GetApp("external-service-in-kube-zone1").GetIP(), "external-service-in-kube-zone1", "kuma-1-zone"))).
+			Install(YamlUniversal(externalService(mesh, env.UniZone1.GetApp("external-service-in-both-zones").GetIP()))).
+			Install(YamlUniversal(zoneExternalService(meshNoZoneEgress, env.UniZone1.GetApp("external-service-in-uni-zone4").GetIP(), "demo-es-in-uni-zone4", "kuma-4"))).
+			Setup(env.Global)).ToNot(HaveOccurred())
+	})
+
+	BeforeEach(func() {
+		Eventually(func(g Gomega) {
+			g.Expect(env.UniZone1.GetZoneEgressEnvoyTunnel().ResetCounters()).To(Succeed())
```

Comment:
does it work consistently? Looking at the code I think that tunnels are set only by the main test process, but in other processes it will be empty